auraed/src/runtime/mod.rs

```diff
@@ -237,29 +192,128 @@ impl cell_service_server::CellService for CellService {
                 err: e.to_string(),
             }
         })?;
+
         let child_id = child.id();
         info!(
             "CellService: stop() cell_name={:?} executable_name={:?} pid={child_id}",
-            &cell_name,
-            &executable_name,
+            cell_name,
+            executable_name,
         );
+
         // TODO: check for
         child.kill().map_err(|e| CellServiceError::Internal {
             msg: format!("failed to kill child with pid {child_id}"),
             err: e.to_string(),
         })?;
+
         let exit_status =
             child.wait().map_err(|e| CellServiceError::Internal {
                 msg: format!("failed to wait for child with pid {child_id}"),
                 err: e.to_string(),
             })?;
+
         info!(
             "Child process with pid {child_id} exited with status {exit_status}",
         );
 
-        // Ok
         Ok(Response::new(StopCellResponse {}))
     }
+
+    // Here is where we define the "default" cgroup parameters for Aurae cells
+    fn create_cgroup(&self, cell: ValidatedCell) -> Result<Cgroup, Error> {
+        let ValidatedCell {
```

Comment:
I'm not sure I am understanding this question, so I'll answer it a couple of ways:

If the concern is the destructuring of `ValidatedCell` (i.e., `let ValidatdCell { all, the, props, is, a, lot } = cell;`) and typing out all those props, I (using clion, but I think it works in VSCode too) don't actually type any of the fields. All I type is `let ValidatedCell {} = cell;` and then the IDEs quick fix takes care of the rest. I know that's not helpful if someone uses an IDE that doesn't do it for them, but there are other benefits to destructuring. Specifically, in this function, I was able to pass the value of each field into `CgroupBuilder` without needing to clone anything.

If you are asking about converting a `Cell` type to a `ValidatedCell` type, that happens in the validation step. Most of it is generated. If you expand the `ValidatedType` macro attribute on top of `ValidatedCell` you get:

```rust
impl ::validation::ValidatedType<Cell> for ValidatedCell {
    fn validate(input: Cell, parent_name: Option<&str> ) -> ::std::result::Result<ValidatedCell, ::validation::ValidationError> {
        CellValidator::validate_fields(&input, parent_name)?;
        let Cell { name, cpu_cpus, cpu_shares, cpu_mems, cpu_quota, } = input;
        let name = CellValidator::validate_name(name, "name", parent_name)?;
        let cpu_cpus = CellValidator::validate_cpu_cpus(cpu_cpus, "cpu_cpus", parent_name)?;
        let cpu_shares = CellValidator::validate_cpu_shares(cpu_shares, "cpu_shares", parent_name)?;
        let cpu_mems = CellValidator::validate_cpu_mems(cpu_mems, "cpu_mems", parent_name)?;
        let cpu_quota = CellValidator::validate_cpu_quota(cpu_quota, "cpu_quota", parent_name)?;
        Ok(ValidatedCell { name, cpu_cpus, cpu_shares, cpu_mems, cpu_quota })
    }
}
trait CellTypeValidator {
    fn validate_name(name: String, field_name: &str, parent_name: Option<&str>) -> ::std::result::Result<CellName, ::validation::ValidationError>;
    fn validate_cpu_cpus(cpu_cpus: String, field_name: &str, parent_name: Option<&str>) -> ::std::result::Result<String, ::validation::ValidationError> { Ok(cpu_cpus) }
    fn validate_cpu_shares(cpu_shares: u64, field_name: &str, parent_name: Option<&str>) -> ::std::result::Result<u64, ::validation::ValidationError> { Ok(cpu_shares) }
    fn validate_cpu_mems(cpu_mems: String, field_name: &str, parent_name: Option<&str>) -> ::std::result::Result<String, ::validation::ValidationError> { Ok(cpu_mems) }
    fn validate_cpu_quota(cpu_quota: i64, field_name: &str, parent_name: Option<&str>) -> ::std::result::Result<i64, ::validation::ValidationError> { Ok(cpu_quota) }
    fn validate_fields(fields: &Cell, parent_name: Option<&str>) -> ::std::result::Result<(), ::validation::ValidationError> { Ok(()) }
}
struct CellValidator;
```  

The above code does:
1. Generate a `validate` function that destructures `Cell` and calls a validation function for each field.
2. The field validation functions are generated next. In this case the `CellTypeValidator` trait. Because I decorated most of the fields with `#[validate(none)]` most of them have pass through default implementations. `validate_name` is the only function that requires an implementation. If I had added `#[validate]` to the `name` field, I would get `validation::ValidatedField::validate(Some(name), field_name, parent_name)` which is the same as calling `CellName::validate`. There is also `#[validate(opt)]` which would call `CellName::validate_optional`.
3. It creates an empty struct to implement the `TypeValidator` trait on. It needs to create it to have control over the name of the struct so that it knows how to call it. 

An additional note, if all the fields are able to to be decorated with either `#[validate]`, `#[validate(opt)]`, or `#[validate(none)]` the `TypeValidator` trait would not have any functions requiring implementations, which lead to `impl CellTypeValidator for CellValidator {}` which I know you noted before as being a little weird, but it's just the result of the macro taking care of everything else.  Perhaps I should consider having the macro also generate that last bit in the event all the fields validation functions have default implementations.